### PR TITLE
lint: deny unsafe code & setup dependabot

### PR DESCRIPTION
**Pull request by Mathieu-Lala**
* setup dependabot (see https://docs.github.com/en/code-security/dependabot)
* set `#![deny(unsafe_code)]` at top level of crates and `#[allow(unsafe_code)]` when required

### CODE REVIEW

File: vsmtp-common/src/lib.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 #![doc(html_no_source)]
 #![deny(missing_docs)]
+#![deny(unsafe_code)]
```
Comment (ltabis) [inline vsmtp-common/src/lib.rs:5]: What's the point ?

**Review by ltabis — CHANGES_REQUESTED** [review]

File: vsmtp-common/src/lib.rs (hunk 1 above)
Comment (Mathieu-Lala) [reply vsmtp-common/src/lib.rs:5]: Explicit our no unsafe policy unless required

Comment (ltabis) [reply vsmtp-common/src/lib.rs:5]: I think it's useless

**Review by ltabis — CHANGES_REQUESTED** [review]
remove `#![deny(unsafe_code)]`

**Pull request closed without merging**